app/controllers/mechanic_controller.rb

```diff
@@ -0,0 +1,12 @@
+class MechanicController < ApplicationController
+  
+  def show 
+    @mechanic = Mechanic.find(params[:id])
+  end
+
+  def new 
+    @mechanic = Mechanic.find(params[:id])
+  end
```

Comment:
While this is not exactly "incorrect" the US did not ask for a new view. It wanted the form to be on the show page itself. When reading the US, it is important to follow it exactly. You can compare it to being on the job and being asked to implement a feature and you do your own thing, it probably wouldn't go well. Stick to the US.